graviti/utility/collections.py

```diff
@@ -430,6 +413,32 @@ def _setitem(self, key: str, value: _V) -> None:
             self._keys.append(key)
         self._data.__setitem__(key, value)
 
+
+class NameOrderedDict(MutableMapping[str, _V], InmutableNameOrderedDict[_V]):
```

Comment:
in python builtin collections library, the mutable class will have `Mutable` prefix, like `MutableMapping`, and the Immutable class will have no prefix, like `Mapping`. I think we should follow this convention.

Reply:
Here `Dict` means `Mutable`, so that in collections we have `UserDict`, `OrderedDict` they are all mutable,
here there is another naming convention for `immutable dict` which is `frozen dict`, so I think we can use `FrozenNameOrderedDict` for what you need.

Reply:
here is some reference:
https://docs.python.org/zh-cn/3/library/stdtypes.html?highlight=frozenset#frozenset
https://peps.python.org/pep-0416/